src/include/units/quantity_cast.h

```diff
@@ -220,10 +221,15 @@ template<QuantityValue ToRep, typename D, typename U, scalable_with_<ToRep> Rep>
  *
  * Implicit conversions between quantity points of different types are allowed only for "safe"
  * (i.e. non-truncating) conversion. In other cases an explicit cast has to be used.
+ * If the cast re-references the origin in the quantity point, the distance between the origins must be known
+ * and fixed; the value will the correctly be adjusted to refer to the same physical point.
  *
- * This cast gets the target quantity point type to cast to or anything that works for quantity_cast. For example:
+ * This cast gets the target quantity point type to cast to, the new point origin or anything that works for quantity_cast.
+ * For example:
  *
  * auto q1 = units::quantity_point_cast<decltype(quantity_point{0_q_s})>(quantity_point{1_q_ms});
+ * auto q1 = units::quantity_point_cast<decltype(0_qp_deg_F)>(1_qp_deg_C);
+ * auto q1 = units::quantity_point_cast<units::physical::si::celsius_temperature_origin>(1_qp_deg_F);
```

Comment:
I thought that we are not providing UDLs for quantity points in the library (https://mpusz.github.io/units/framework/quantity_points.html#differences-to-quantity)?

Reply:
Well, I added that because you said this: https://github.com/mpusz/units/issues/1#issuecomment-557810476.

Reply:
Yeah, and it still stands. Although, I must admit that `1_q_deg_C` and `1_qp_deg_C` look pretty distinct to me. Unit constants are going to prove worse here... 😞

Reply:
Maybe it does now. But the difference is just 1 letter. I switched to `l` and `r` from `lhs` and `rhs` due to how often I confused them.

Reply:
This is why I think that quantity_points should be explicit in the code. Most regular users will have problems in understanding the difference. Also, UDLs are only for compile-time constants. The more I work with the library the more I think that unit constants is the way to go (https://mpusz.github.io/units/framework/quantities.html#udls-vs-unit-constants) and I do not see an easy way to make quantity points distinct enough here.

Reply:
First I thought that quantity constants do not work for quantity points, because they cannot be multiplied. However, Now that I think about it, I think there is an elegant solution: `1*m + mean_sea_level`!

Reply:
The other thing that could be considered would be to use CTAD again to let `quantity_point(1 * deg_C)` magically pick the right origin point, here `celsius_temperature_origin`. I don't like that because it would be far too risky for the magic kicking in in generic code, leading to silently wrong origins being picked.

Reply:
The more I think about `quantity + origin` as a generic way of constructing a quantity point, the more I like it :-). With that, the need for other methods to construct quantity points is less pressing, and I leave it up to you/the general UDL discussion if we should provide them literals for quantity points, and how they should be named.

Reply:
I like that. It's a very convenient syntax to construct `quantity_point`s.

Reply:
I like it too! 👍